modules/aerodyn/src/AeroDyn.f90

```diff
@@ -1639,7 +1662,52 @@ subroutine AD_UpdateStates( t, n, u, utimes, p, x, xd, z, OtherState, m, errStat
    subroutine Cleanup()
       call AD_DestroyInput( uInterp, errStat2, errMsg2)
    end subroutine Cleanup
+   logical function Failed()
+      call SetErrStat(errStat2, errMsg2, errStat, errMsg, 'AD_CalcWind')
+      Failed = errStat >= AbortErrLev
+      if (Failed) call Cleanup()
+   end function Failed
 end subroutine AD_UpdateStates
+
+subroutine AD_CalcWind(t, u, p, o, m, ErrStat, ErrMsg)
+
+   real(DbKi),                intent(in   )     :: t        !< Current simulation time in seconds
+   type(AD_InputType),        intent(inout)     :: u        !< Inputs at Time t
+   type(AD_ParameterType),    intent(in   )     :: p        !< Parameters
+   type(AD_OtherStateType),   intent(in   )     :: o        !< Other states at t
+   type(AD_MiscVarType),      intent(inout)     :: m        !< Misc/optimization variables
+   integer(IntKi),            intent(  out)     :: ErrStat  !< Error status of the operation
+   character(*),              intent(  out)     :: ErrMsg   !< Error message if ErrStat /= ErrID_None
+   
+   integer(intKi)                               :: ErrStat2
+   character(ErrMsgLen)                         :: ErrMsg2
+   integer(intKi)                               :: node
+
+   ErrStat = ErrID_None
+   ErrMsg = ""
+   
+   ! Set wind positions
+   node = 0
+   call AD_SetExternalWindPositions(p, u, o, m%WindPos, node, errStat2, errMsg2)
+   if(Failed()) return
+
+   ! Calculate wind velocity/acceleration
+   call IfW_FlowField_GetVelAcc(p%FlowField, 1, t, m%WindPos, &
+                                m%WindVel, m%WindAcc, errStat2, errMsg2)
+   if(Failed()) return
+
+   ! Transfer wind velocity to nodes
+   node = 1
```

Comment:
Yes, I'll add a comment since it was really confusing the first time I looked at it. The code that set's the positions increments the node number fore setting the value, and the code that sets the wind, increments the node position after setting the value. It would be good to make it consistent.